src/coreclr/jit/morph.cpp

```diff
@@ -3340,7 +3340,7 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
 
                         if (argLclNum != BAD_VAR_NUM)
                         {
-                            argObj->ChangeType(argVarDsc->TypeGet());
+                            argx->ChangeType(argVarDsc->TypeGet());
```

Comment:
This was wrong for commas, we were changing the type of the effective node but not the entire comma tree.